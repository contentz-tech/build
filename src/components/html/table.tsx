import { jsx } from "@emotion/core";

function Table(props: any) {
  return (
    <table
      {...props}
      css={{
        fontSize: "1em",
        width: "100%"
      }}
    />
  );
}

function TH(props: any) {
  return (
    <th
      {...props}
      css={{
        padding: 5,
        color: "#9b9b9b",
        fontStyle: "oblique",
        fontWeight: "normal",
        textAlign: "left"
      }}
    />
  );
}

function TR(props: any) {
  return <tr {...props} css={{ padding: 5 }} />;
}

function TD(props: any) {
  return <td {...props} css={{ padding: 5 }} />;
}

export { Table, TH, TR, TD };
