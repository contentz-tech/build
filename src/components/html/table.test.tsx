/** * @jest-environment jsdom */
import '@testing-library/jest-dom/extend-expect'
import { jsx } from "@emotion/core";
import { render } from "@testing-library/react";
import { Table, TH, TR, TD } from "./table";

describe("Table", () => {
  test("Table without Header", () => {
    const { asFragment } = render(
      <Table>
        <tbody>
          <TR>
            <TD>Sergio</TD>
            <TD>Xalambrí</TD>
          </TR>
        </tbody>
      </Table>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("Table with Header", () => {
    const { asFragment } = render(
      <Table>
        <thead>
          <TR>
            <TH>First name</TH>
            <TH>Last name</TH>
          </TR>
        </thead>
        <tbody>
          <TR>
            <TD>Sergio</TD>
            <TD>Xalambrí</TD>
          </TR>
        </tbody>
      </Table>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
