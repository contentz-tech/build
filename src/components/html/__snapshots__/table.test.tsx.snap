// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Table with Header 1`] = `
<DocumentFragment>
  .emotion-6 {
  font-size: 1em;
  width: 100%;
}

.emotion-2 {
  padding: 5px;
}

.emotion-3 {
  padding: 5px;
}

.emotion-0 {
  padding: 5px;
  color: #9b9b9b;
  font-style: oblique;
  font-weight: normal;
  text-align: left;
}

<table
    class="emotion-6"
  >
    <thead>
      <tr
        class="emotion-2"
      >
        <th
          class="emotion-0"
        >
          First name
        </th>
        <th
          class="emotion-0"
        >
          Last name
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="emotion-2"
      >
        <td
          class="emotion-3"
        >
          Sergio
        </td>
        <td
          class="emotion-3"
        >
          Xalambrí
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Table Table without Header 1`] = `
<DocumentFragment>
  .emotion-3 {
  font-size: 1em;
  width: 100%;
}

.emotion-2 {
  padding: 5px;
}

.emotion-0 {
  padding: 5px;
}

<table
    class="emotion-3"
  >
    <tbody>
      <tr
        class="emotion-2"
      >
        <td
          class="emotion-0"
        >
          Sergio
        </td>
        <td
          class="emotion-0"
        >
          Xalambrí
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
